File: algorithm_leetcode/src/main/java/com/huldar/java/day01/ReverseInterger.java
package com.huldar.java.day01;

/**
 * LeetCode(7):整数反转
 *
 * @author huldar
 * @date 2019/6/4 11:26
 */
public class ReverseInterger {
    /**
     * @param x
     * @return
     */
    public static int reverse(int x) {
        int res = 0;
        while (x != 0) {
            //取出该个位的数字
            int num = x % 10;
            //数字向左移动一位 末尾去掉
            x = x / 10;
            if (res > Integer.MAX_VALUE / 10 || (res == Integer.MAX_VALUE / 10 && num > 7)) {
                return 0;
            }
            if (res < Integer.MIN_VALUE / 10 || (res == Integer.MIN_VALUE / 10 && num < -8)) {
                return 0;
            }
            res = res * 10 + num;
        }
        return res;
    }
}

File: algorithm_leetcode/src/main/scala/com/huldar/scala/day01/ReverseInterger.scala
package com.huldar.scala.day01

/**
  *
  * 数反转
  *
  * @author huldar
  *
  */
class ReverseInterger {

  def reverse(tag: Int): Int = {
    var res = 0
    var num = 0
    var tag1 = tag
    while (tag1 != 0) {
      num = tag1 % 10
      res = res * 10 + num
      if (res > Int.MaxValue || res < Int.MinValue) {
        return 0
      }
      tag1 = tag1 / 10
    }
    res
  }

  def reverse1(tag: Int): Int = {
    if (tag > -10 && tag < 10)
      tag
    else {
      try {
        val absTag = Math.abs(tag)
        tag / absTag * absTag.toString.reverse.toInt
      } catch {
        case _: Throwable => 0
      }
    }

  }

}
